src/Sidebar/Menu/Submenu/index.jsx

```diff
@@ -1,17 +1,50 @@
-import React from 'react';
-import Icon from './Icon';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ArrowIcon from './Icon';
 import { Nav, SubmenuLink, SubmenuTitle } from './styled';
 
-export default () => (
-  <div className="submenu">
-    <div className="submenu-title active">
-      <SubmenuTitle isActive>sports</SubmenuTitle>
-      <Icon />
-    </div>
-    <Nav>
-      <SubmenuLink isActive>shoes</SubmenuLink>
-      <SubmenuLink>clothing</SubmenuLink>
-      <SubmenuLink>accesories</SubmenuLink>
-    </Nav>
-  </div>
-);
+class SubMenu extends Component {
+  constructor(props) {
+    super(props);
+    this.toggleState = this.toggleState.bind(this);
+    this.state = {
+      isOpened: false,
+    };
+  }
+
+  toggleState() {
+    this.setState({ isOpened: !this.state.isOpened });
```

Comment:
https://medium.freecodecamp.com/functional-setstate-is-the-future-of-react-374f30401b6b
https://medium.com/@shopsifter/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1